sno/fast_import.py

```diff
@@ -104,8 +105,57 @@ def fast_import_tables(
                 raise ValueError(f"{blob_path} already exists")
 
         for source in sources:
+            replacing_dataset = None
             if replace_existing == ReplaceExisting.GIVEN:
                 p.stdin.write(f"D {source.dest_path}\n".encode('utf8'))
+
+                # Optimisation: Try to avoid rewriting features for compatible schema changes.
+                # i.e. we can avoid rewriting feature blobs if the only schema changes are:
+                #  * dropped columns
+                #  * renamed columns
+                #  * reordered columns
+                try:
+                    replacing_dataset = RepositoryStructure(repo)[source.dest_path]
+                except KeyError:
+                    pass
+                else:
+                    # If the schemas are *identical, there's no need to pass this dataset object on.
+                    # Features won't be rewritten anyway.
+                    if replacing_dataset.schema == source.schema:
+                        replacing_dataset = None
+                    else:
+                        # Check that the schemas are compatible.
```

Comment:
Look at Schema.diff_type_counts - that has some code that categorises schema changes.

You could use it here, or you could add another function to Schema that does the check that you care about.